src/fireedge/src/modules/components/Charts/MultiChart/helpers/subComponents/PolarChart.js

```diff
@@ -0,0 +1,144 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2025, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+import { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Box, Paper, Typography } from '@mui/material'
+import { isDevelopment } from '@UtilsModule'
+import { Tr } from '@modules/components/HOC'
+/**
+ * Formats the input data for use in a polar chart.
+ *
+ * @param {Array|object} input - The data to be formatted.
+ * @param {number} datasetIdx - Index of the dataset to format
+ * @returns {Array} The formatted dataset.
+ */
+export const FormatPolarDataset = (input, datasetIdx = 0) => {
+  const logError = (message) => {
+    if (isDevelopment()) console.error(`FormatPolarDataset: ${message}`)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when untrusted user input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input using `encodeURIComponent()` or `DOMPurify.sanitize()` before logging.  Learn more - https://cwe.mitre.org/data/definitions/117.html